src/nfa/nfa-complement.cc

```diff
@@ -37,9 +37,9 @@ Nfa Mata::Nfa::Algorithms::complement_classical(
 	}
 
 	result = determinize(aut, subset_map);
-	State sink_state = result.states_number() + 1;
+	State sink_state = result.delta.post_size() + 1;
 	result.increase_size(sink_state+1);
-	assert(sink_state < result.states_number());
+	assert(sink_state < result.delta.post_size());
```

Comment:
Looking at how `Delta::post_size()` is used, I really am not a fan of the name for when you want to *get the number of automata states*. `post_size()` makes sense in the context of Delta itself, but not for this often used automata operation. I would not change the name inside Delta.

I would consider, however, adding an *alias* method on the whole NFA `NFA::states_number()` which would return `this.delta.post_size()`, simply to allow the user to type `state num` and find the method. Similarly, there could be `Nfa::transitions_number()` for transitions. But, this goes against our idea that we want to force the user to use the underlying classed for delta, initial and final states. Any ideas?

Reply:
We should discuss this on thursday meeting.

Reply:
I also did not expect that it would be used everywhere. I thought we would use nfa.max_state(), or what the name if that is, everywhere, and use post_size() very seldom, essentially only for determining the nfa.max_state().

Reply:
Ok, i'm gonna check where can be `post_size` hidden a replaced by `max_state()`

Reply:
Well, my IDE shows around 90 usages of `post_size`. I would rather make refactoring to using `max_state` in another PR.